src/dials/algorithms/scaling/scaling_library.py

```diff
@@ -372,8 +374,103 @@ def determine_best_unit_cell(experiments):
     return best_unit_cell
 
 
+class ExtendedDatasetStatistics(iotbx.merging_statistics.dataset_statistics):
+
+    """A class to extend iotbx merging statistics."""
+
+    def __init__(self, *args, additional_stats=False, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.r_split = None
+        self.r_split_binned = None
+        self.binner = None
+        if not additional_stats:
+            return
+        i_obs = kwargs.get("i_obs")
+        n_bins = kwargs.get("n_bins", 20)
+        if not i_obs:
+            return
+        i_obs_copy = i_obs.customized_copy()
+        i_obs_copy.setup_binner(n_bins=n_bins)
+        i_obs = i_obs.map_to_asu()
+        i_obs = i_obs.sort("packed_indices")
+
+        seed = 0
+
+        split_datasets = split_unmerged(
```

Comment:
As I understand it this is randomly assigning the observations to each half dataset (as per CC1/2 calculation) rather than randomly assigning whole images to each half dataset (as per the [published definition](https://journals.iucr.org/j/issues/2012/02/00/db5097/index.html)). I guess in practice it probably doesn't make much difference, but it's probably worth explicitly pointing out where we differ from the published definition, at least by a comment in the code.

Reply:
Thanks, I hadn't appreciated that the published definition is based on whole image splitting. I'm happy that using the random observation assignment, like in cc1/2, is a good way for measuring internal dataset consistency, but I will add a comment.